src/findings_data_import.py

```diff
@@ -49,12 +49,292 @@ def move_processed_file(s3_client, bucket, folder, filename):
+        )
+        return db
+    except ClientError as client_err:
+        raise Exception(
+            "Unable to fetch database credentials from the SSM", client_err
+        ).with_traceback()
+    # Handle all mongo exceptions the same way..
+    except Exception as err:
+        raise Exception("Unable to connect to the mongo db", err).with_traceback()
+
+
+def download_file(
+    s3_client=None,
+    s3_bucket=None,
+    data_filename=None,
+):
+    """Download a file from a specified s3 bucket, and place it in a temporary file path.
+
+    Parameters:
+    -----------
+    s3_client : S3.Client
+        The AWS S3 Client to retrieve the data file with.
+
+    s3_bucket : str
+        The AWS S3 bucket containing the data file.
+
+    data_filename : str
+        The name of the file containing the data in the S3 bucket
+        above.
+
+    Returns
+    -------
+        string : The file path of the newly created temp file
+        dict   : The JSON data dictionary loaded from downloaded file
+    """
+    logging.info("Retrieving %s from %s...", data_filename, s3_bucket)
+
+    try:
+        # Securely create a temporary file to store the JSON data in
+        temp_file_descriptor, temp_data_filepath = tempfile.mkstemp()
+        # Fetch findings data file from S3 bucket
+        s3_client.download_file(
+            Bucket=s3_bucket, Key=data_filename, Filename=temp_data_filepath
+        )
+
+        # Load data JSON
+        with open(temp_data_filepath) as data_json_file:
+            findings_data = json.load(data_json_file)
+
+        logging.info("JSON data loaded from %s.", data_filename)
+        return temp_data_filepath, findings_data
+    except json.JSONDecodeError as json_err:
+        raise (
+            Exception(
+                f"Unable to decode JSON data for {data_filename}", json_err
+            ).with_traceback()
+        )
+    except ClientError as err:
+        raise (
+            Exception(
+                f"Error downloading file {data_filename} from {s3_bucket} ", err
+            ).with_traceback()
+        )
+
+
+def extract_findings(findings_data, field_map_dict):
+    """
+    Validate and return cleaned/processed finding.
+
+    Parameters
+    ----------
+    findings_data : dict
+        The findings dictionary pulled from a findings JSON file
+
+    field_map_dict: dict
+        The dictionary of replacement rules for field names in findings_data
+
+    Returns
+    -------
+    list : A list of findings objects from findings_data that pass validation
+
+    """
+    valid_findings = []
+
+    # if we (v2) get a lone object, wrap it in a list for compatability
+    if type(findings_data) != list:
+        findings_data = [findings_data]
+
+    # Iterate through data and save each record to the database
+    for index, finding in enumerate(findings_data):
+
+        if not finding or not hasattr(finding, "keys"):
```

Comment:
I think this better illustrates that we're looking for dictionaries.

```suggestion
        if not finding or not isinstance(finding, dict):
```